### PR TITLE
CLN: 29547 replace old string formatting 1

### PR DESCRIPTION
**Pull request by 3vts**
I splitted PR #31844 in batches, this is the first one
For this PR I ran the command `grep -l -R -e '%s' -e '%d' -e '\.format(' --include=*.{py,pyx} pandas/` and checked all the files that were returned for `.format(` and changed the old string format for the corresponding `fstrings` to attempt a full clean of, [#29547](https://github.com/pandas-dev/pandas/issues/29547). I may have missed something so is a good idea to double check just in case

- [ x  ] tests added / passed
- [ x ] passes `black pandas`
- [ x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/frame/test_alter_axes.py [hunk 1]
```diff
@@ -383,7 +383,7 @@ class Thing(frozenset):
             def __repr__(self) -> str:
                 tmp = sorted(self)
                 # double curly brace prints one brace in format string
-                return "frozenset({{{}}})".format(", ".join(map(repr, tmp)))
+                return f"frozenset({{{', '.join(map(repr, tmp))}}})"
```
Comment (MomIsBestFriend) [inline pandas/tests/frame/test_alter_axes.py]: Can you take the "join" outside the f-string?

Comment (3vts) [reply pandas/tests/frame/test_alter_axes.py]: Can you please elaborate more? I did not understand your idea

Comment (WillAyd) [reply pandas/tests/frame/test_alter_axes.py]: `joined_reprs = ', '.join(map(repr, tmp))` should be fine

**Comment by WillAyd** [comment]
lgtm outside of @MomIsBestFriend comment

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Great thanks @3vts

**Pull request merged**